annotation-dereferenciation/src/main/java/eu/europeana/annotation/dereferenciation/MetisDereferenciationClient.java

```diff
@@ -140,46 +144,72 @@ public synchronized Map<String, String> dereferenceOne(String baseUrl, String ur
      */
     public synchronized Map<String, String> dereferenceMany(String baseUrl, List<String> uris, String language) throws AnnotationDereferenciationException {
 	Map<String, String> res = new HashMap<String, String>();
-	String xmlResponse, jsonLdStr;
+	String jsonLdStr;
+	InputStream streamResponse;
 	try {
-        	@SuppressWarnings({ "rawtypes", "unchecked" })
-		String urisJson = JsonSerializer.toString((List)uris);
-	        xmlResponse = getHttpConnection().postRequest(baseUrl, urisJson);
-	        for (Object uri : uris) {
-	            jsonLdStr = convertToJsonLd((String)uri, xmlResponse, language).toString();
-        	    res.put((String)uri, jsonLdStr);
-         	}
+      	@SuppressWarnings({ "rawtypes", "unchecked" })
+      	String urisJson = JsonSerializer.toString((List)uris);
+      	streamResponse = getHttpConnection().postRequest(baseUrl, urisJson);
+      	String[] urisArray = new String[uris.size()];
+        urisArray = uris.toArray(urisArray);
+        jsonLdStr = convertToJsonLd(urisArray, streamResponse, language).toString();
+        //we need to separate the individual jsons manually
+        List<Integer> startingPositions = findSubstringIndexes(jsonLdStr, "{ \"@context\":");
+        for (int i = 0; i < startingPositions.size(); i++) {
+          if(i==startingPositions.size()-1) {
+            res.put(uris.get(i), jsonLdStr.substring(startingPositions.get(i), jsonLdStr.length()));
+          }
+          else {
+            res.put(uris.get(i), jsonLdStr.substring(startingPositions.get(i), startingPositions.get(i+1)));
+          }
+        }          
 	}catch(IOException ex) {
 	    throw new AnnotationDereferenciationException(ex);
 	}catch(RuntimeException ex) {
 	    throw new AnnotationDereferenciationException(ex);
 	}
 	return res;
     }
-
+    
+    /*
+     * TODO: move this function to utils
+     */
+    private List<Integer> findSubstringIndexes(String input, String substring) {
```

Comment:
it is not easy to understand clear what this method is doing, please add a method documentation